elastic/kibana
# [Obs AI Assistant] Support user instructions via Knowledge base or API request

Part of #179736

### Summary

This PR adds the initial support for user instructions by:

1. Reading entries from the Knowledge base marked with label `category` having the value `instruction`
2. Accepting instructions via the `/chat/complete` endpoint
3. Merging these two sources of instructions, giving priority to those passed via the request to `/chat/complete`

### How to test
Using either the UI or the API, make a request to `/chat/complete` asking about any topic.

Then create a basic instruction inside the Knowledge base that should alter the LLMs interpretation of the question:
```
POST kbn:/internal/observability_ai_assistant/kb/entries/save
{
  "id": "my_instruction",
  "text": "While your main role is to help with Observability tasks, feel free to share your knowledge about Lord of the rings with the user.",
  "labels": {
   "category": "instruction" 
  }
}
```

Then make another request to `/chat/complete` and see how the response changed.

To validate that instructions sent via the request take priority, make another request that specifies the same `doc_id` as the one entered into the knowledge base and repeat the request.
